oaknational/oak-ai-lesson-assistant
# refactor: demo account components to oak components - AI-460

## Description

- Refactors demo components to use Oak Components (excluding model)
- Refactors header to use Oak components 
- Fixes margin bug on help page (margin was too small with extra demo banner height)

These changes have had some subtle margin / font size changes due to Oak Components spacing and font type constraints.

## Issue(s)

Fixes #

## How to test

1. Go to https://oak-ai-lesson-assistant-61gyyikpr.vercel-preview.thenational.academy
Toggle isDemoUser true in clerk
Check banner desktop / mobile , check links are working , check pop up for first demo lesson / used all demo lessons up

## Screenshots

How it used to look (delete if n/a):
![image](https://github.com/user-attachments/assets/270bf616-20eb-4d48-aa3f-d3015afe2b8c)
![image](https://github.com/user-attachments/assets/bd1cbf82-411c-4b68-a8e2-bb9b56562226)
![image](https://github.com/user-attachments/assets/2a134161-3886-4ed0-8ccc-6567e3cebb32)
![image](https://github.com/user-attachments/assets/dfff6046-8ef1-43a3-b2a2-9b3d8d735a40)


How it should now look:
![image](https://github.com/user-attachments/assets/7d25cfd2-ca78-45e4-bc8e-bb14f5a54f48)
![image](https://github.com/user-attachments/assets/a95b87d2-6b64-4954-b65c-5888db59895a)
![image](https://github.com/user-attachments/assets/72392019-b21b-4cb4-ab34-1888bfcc3bac)
![image](https://github.com/user-attachments/assets/f2d2efe4-227e-4b2a-92b7-9a9b255eef95)








## Checklist
